tests/Serializer/GraphNavigatorTest.php

```diff
@@ -131,6 +132,37 @@ public function testNavigatorChangeTypeOnSerialization()
         $navigator->accept($object, null);
     }
 
+    public function testExposeAcceptHandlerExceptionOnSerialization()
+    {
+        $object = new SerializableClass();
+        $typeName = 'JsonSerializable';
+        $format = 'foo';
+        $msg = 'Useful serialization error with relevant context information';
+
+        $handler = static function ($visitor, $data, array $type, SerializationContext $context) use ($msg) {
+            $context->startVisiting(new \stdClass());
+            throw new \RuntimeException($msg);
+        };
+        $this->handlerRegistry->registerHandler(GraphNavigatorInterface::DIRECTION_SERIALIZATION, $typeName, $format, $handler);
+
+        $visitor = $this->createMock(VisitorInterface::class);
```

Comment:
can we avoid all this mocks and use existing `$this->serializationVisitor` and `$this->context` objects?

Reply:
Yes, but because it is testing the `startVisiting` and `stopVisiting` internal methods I will have to change which methods are mocked on the `$this-context` and add the `initialize()` call to the `testNavigatorChangeTypeOnSerialization`.

Reply:
:+1: